spec/models/post_mover_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require_relative "../fabricators/reaction_fabricator.rb"
+require_relative "../fabricators/reaction_user_fabricator.rb"
+
+describe PostMover do
+  fab!(:admin) { Fabricate(:admin) }
+  fab!(:user) { Fabricate(:user) }
+  fab!(:topic_1) { Fabricate(:topic, user: user) }
+  fab!(:topic_2) { Fabricate(:topic, user: user) }
+  fab!(:topic_3) { Fabricate(:topic, user: user) }
+  fab!(:post_1) { Fabricate(:post, topic: topic_1, user: user) }
+  fab!(:post_2) { Fabricate(:post, topic: topic_1, user: user) }
+  fab!(:reaction_1) { Fabricate(:reaction, post: post_1, reaction_value: "clap") }
+  fab!(:reaction_2) { Fabricate(:reaction, post: post_1, reaction_value: "confetti") }
+  fab!(:reaction_3) { Fabricate(:reaction, post: post_2, reaction_value: "heart") }
+  fab!(:reaction_4) { Fabricate(:reaction, post: post_2, reaction_value: "wave") }
+  fab!(:user_reaction_1) { Fabricate(:reaction_user, reaction: reaction_1, post: post_1) }
+  fab!(:user_reaction_2) { Fabricate(:reaction_user, reaction: reaction_2, post: post_1) }
+  fab!(:user_reaction_3) { Fabricate(:reaction_user, reaction: reaction_3, post: post_2) }
+  fab!(:user_reaction_4) { Fabricate(:reaction_user, reaction: reaction_4, post: post_2) }
+
+  before { SiteSetting.discourse_reactions_enabled = true }
+
+  it "new post has topic's first post reactions (OP)" do
```

Comment:
Minor one but rspec descriptions are supposed to read like a proper sentence. Perhaps something like this? 

```suggestion
  it "should add old post's reactions to new post when a topic's first post is moved" do
```